packages/cli-lib/lib/watch.js

```diff
@@ -42,64 +40,57 @@ const _notifyOfThemePreview = (filePath, accountId) => {
 const notifyOfThemePreview = debounce(_notifyOfThemePreview, 1000);
 
 async function uploadFile(accountId, file, dest, options) {
-  const src = yargs.argv.src;
-  const fieldOptions = options.fieldOptions;
-  if (!isAllowedExtension(file)) {
+  const src = options.src;
+  const processFieldsJs = isProcessableFieldsJs(src, file);
+
+  if (!isAllowedExtension(file) && !processFieldsJs) {
     logger.debug(`Skipping ${file} due to unsupported extension`);
     return;
   }
   if (shouldIgnoreFile(file)) {
     logger.debug(`Skipping ${file} due to an ignore rule`);
     return;
   }
-  const processFieldsJsOpt = yargs.argv.processFieldsJs;
-  const processFieldsJs =
-    isProcessableFieldsJs(src, file) && processFieldsJsOpt;
-  let fieldsJs;
   if (processFieldsJs) {
-    fieldsJs = new FieldsJs(src, file, undefined, fieldOptions);
-    const outputPath = await fieldsJs.getOutputPathPromise();
-    if (fieldsJs.rejected) return;
+    const parsedOptions = JSON.stringify({
+      ...options.commandOptions,
+      src: file,
+      dest: path.join(path.dirname(dest), 'fields.json'),
+      mode: options.mode,
+      _: ['upload'],
+    });
 
-    fieldsJs.outputPath = outputPath;
-    dest = path.join(path.dirname(dest), 'fields.json');
+    const fieldsJsProcessCode = `require('@hubspot/cli/commands/upload').handler(${parsedOptions})`;
+    spawn(fieldsJsProcessCode);
```

Comment:
Just to make it clear what is going on here and why I'm doing things this way...

[So as it stands, ES modules cannot be removed from the module cache](https://github.com/nodejs/modules/issues/307). This is not a big deal for `hs upload` because once the module is loaded in and uploaded, the process dies and the cache is cleared. But for `hs watch`, since the process persists, if we do not clear previously loaded JS field modules out, then old ones will be loaded when a user changes a file. Thus, `fields.js` files will never update while being watched.

So what happens here is that whenever we are running a watch and encounter a processable JS fields file,  we spawn a child node process and call the `hs upload` command. The nice thing here is that since child processes have their own memory, and die immediately after running, we get a clear cache every time. There is a small performance hit with this, but seems negligible. 

There are some other workarounds to this (Transform ESM to CJS with Babel, or [doing something like this](https://github.com/nodejs/modules/issues/307#issuecomment-764560656)), but both have pretty big issues. This solution (courtesy @timmfin) works really well.

Reply:
It may be worth adding in a short comment to briefly explain some of this since it's not immediately apparent in the code.